setools/diff/terules.py

```diff
@@ -38,32 +42,178 @@
 modified_terule_record = namedtuple("modified_terule", ["rule", "added_default", "removed_default"])
 
 
-def _avrule_expand_generator(rule_list, WrapperClass):
-    """
-    Generator that yields wrapped, expanded, av(x) rules with
-    unioned permission sets.
-    """
-    items = dict()
+class Side(Enum):
+    left = 0
+    right = 1
 
-    for unexpanded_rule in rule_list:
-        for expanded_rule in unexpanded_rule.expand():
-            expanded_wrapped_rule = WrapperClass(expanded_rule)
 
-            # create a hash table (dict) with the first rule
-            # as the key and value.  Rules where permission sets should
-            # be unioned together have the same hash, so this will union
-            # the permissions together.
-            try:
-                items[expanded_wrapped_rule].perms |= expanded_wrapped_rule.perms
-            except KeyError:
-                items[expanded_wrapped_rule] = expanded_wrapped_rule
+rule_db_side_data_record = namedtuple("rule_db_side_data", ["perms", "orig_rule"])
 
-    if items:
-        logging.getLogger(__name__).debug(
-            "Expanded {0.ruletype} rules for {0.policy}: {1}".format(
-                unexpanded_rule, len(items)))
+rule_db_sides_record = namedtuple("rule_db_sides", ["left", "right"])
```

Comment:
Please move the namedtuple decls up with the other ones.

Reply:
I moved type_db_record up.